app/Models/User.php

```diff
@@ -284,4 +288,76 @@ public function preferredLocale(): string {
+            unset($webhook['url']);
+            return $webhook;
+        });
+
+
+        if ($this->avatar && file_exists(public_path('/uploads/avatars/' . $this->avatar))) {
+            $personalDataSelection
+                ->addFile(public_path('/uploads/avatars/' . $this->avatar));
+        }
+
+        $personalDataSelection
+            ->add('user.json', $user)
+            ->add('statuses.json', $this->statuses()->with('tags')->get())
+            ->add('notifications.json', $this->notifications()->get()->toJson())
+            ->add('likes.json', $this->likes()->get()->toJson())
+            ->add('social_profile.json', $this->socialProfile()->with('mastodonserver')->get())
+            ->add('event_suggestions.json', EventSuggestion::where('user_id', $this->id)->get()->toJson())
+            ->add('events.json', Event::where('approved_by', $this->id)->get()->toJson())
+            ->add('webhooks.json', $webhooks)
+            ->add(
+                'webhook_creation_requests.json',
+                WebhookCreationRequest::where('user_id', $this->id)->get()->toJson()
+            )
+            ->add('tokens.json', TokenController::index($this)->toJson())
+            ->add('ics_tokens.json', $this->icsTokens()->get()->toJson())
+            ->add(
+                'password_resets.json',
+                DB::table('password_resets')->select(['email','created_at'])->where('email', $this->email)->get()
+            )
+            ->add('apps.json', $this->oAuthClients()->get()->toJson())
+            ->add('follows.json', DB::table('follows')->where('user_id', $this->id)->get())
+            ->add('followings.json', DB::table('follows')->where('follow_id', $this->id)->get())
+            ->add('blocks.json', DB::table('user_blocks')->where('user_id', $this->id)->get())
+            ->add('blocked_by.json', DB::table('user_blocks')->where('blocked_id', $this->id)->get())
+            ->add('mutes.json', DB::table('user_mutes')->where('user_id', $this->id)->get())
+            ->add('muted_by.json', DB::table('user_mutes')->where('muted_id', $this->id)->get())
+            ->add('follow_requests.json', DB::table('follow_requests')->where('user_id', $this->id)->get())
+            ->add('follows_requests.json', DB::table('follow_requests')->where('follow_id', $this->id)->get())
+            ->add('sessions.json', $this->sessions()->get()->toJson())
+            ->add('home.json', $this->home()->get()->toJson())
+            ->add('hafas_trips.json', DB::table('hafas_trips')->where('user_id', $this->id)->get())
+            ->add('mentions.json', Mention::where('user_id', $this->id)->get()->toJson())
+            ->add('roles.json', $this->roles()->get()->toJson())
+            ->add(
+                'activity_log.json',
+                DB::table('activity_log')->where('causer_type', get_class($this))->where('causer_id', $this->id)->get()
+            )
+            ->add('permissions.json', $this->permissions()->get()->toJson());
```

Comment:
Ich finde es langfristig gesehen etwas kritisch, wenn wir einfach DB Queries raw ausspucken. Später hinzugefügte Columns, die nicht exposed werden sollten (z.B. irgendwelche Keys o.ä.) werden hier sicher vergessen. Wäre es nicht sinnvoller das noch einmal durch die Resourcen zu jagen?

Reply:
Alternativ einfach ein `->select([...])` mit den gewünschten Columns, damit nichts falsches exposed wird.

Reply:
In Arbeit mit #3009